testing/citest/tests/azure_smoke_test.py

```diff
@@ -0,0 +1,303 @@
+                },
+                'user': '[anonymous]'
+            }],
+            description= 'Test - delete application {app}'.format(app=self.bindings['TEST_APP']),
+            application=self.bindings['TEST_APP'])
+
+    builder = az.AzContractBuilder(self.az_observer)
+    (builder.new_clause_builder(
+        'Application Created', retryable_for_secs=30)
+      .collect_resources(
+          az_resource='storage',
+          command='blob',
+          args=['exists', '--container-name', 'front50',
+          '--name', 'applications/'+self.bindings['TEST_APP']+'/application-metadata.json',
+          '--account-name', self.bindings['azure_storage_account_name'],
+          '--account-key', self.bindings['spinnaker_azure_storage_account_key']])
+      .contains_path_eq('exists',False))
+
+    return st.OperationContract(
+        self.new_post_operation(
+            title='delete_app', data=payload,
+            path='tasks'),
+        contract=builder.build())
+
+
+  def create_a_security_group(self):
+    """Creates AzContract for createServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was created.
+    This will create a Network Security Group in a Resource Group on your azure Subscription
+    """
+    rules = [
+        {
+            "access": "Allow",
+            "destinationAddressPrefix": "*",
+            "destinationPortRange": "80-80",
+            "direction": "InBound",
+            "endPort": 80,
+            "name": self.TEST_SECURITY_GROUP_RULE_1,
+            "priority": 100,
+            "protocol": "tcp",
+            "sourceAddressPrefix": "*",
+            "sourcePortRange": "*",
+            "startPort": 80
+        }]
+    job = [{
+        "provider": "azure",
+        "application": self.bindings['TEST_APP'],
+        "appName": self.bindings['TEST_APP'],
+        "region": self.bindings['TEST_AZURE_RG_LOCATION'],
+        "stack": self.bindings['TEST_STACK'],
+        "description": "Test - create security group for {app}".format(
+            app=self.bindings['TEST_APP']),
+        "detail": "",
+        "credentials": self.bindings['SPINNAKER_AZURE_ACCOUNT'],
+        "securityRules": rules,
+        "name": self.TEST_SECURITY_GROUP,
+        "securityGroupName": self.TEST_SECURITY_GROUP,
+        "cloudProvider": "azure",
+        "type": "upsertSecurityGroup",
+        "user": "[anonymous]"
+    }]
+    builder = az.AzContractBuilder(self.az_observer)
+    (builder.new_clause_builder(
+        'Security Group Created', retryable_for_secs=30)
+    .collect_resources(
+         az_resource='network',
+         command='nsg',
+         args=['show', '--name', self.TEST_SECURITY_GROUP, '--resource-group', self.TEST_SECURITY_GROUP_RG],
+         no_resources_ok=True)
+    #sec grp name matches expected
+    .contains_match({'name': jp.STR_SUBSTR(self.TEST_SECURITY_GROUP)})
+    .contains_path_pred(
```

Comment:
This looks older than our conversation earlier today.
I'm going to stop here and ping you because I think this is out of date.